mock-tikv/src/store.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use kvproto::kvrpcpb::KvPair;
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
+};
+
+#[derive(Debug, Clone)]
+pub struct KvStore {
+    data: Arc<RwLock<HashMap<Vec<u8>, Vec<u8>>>>,
+}
+
+impl Default for KvStore {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl KvStore {
+    pub fn new() -> KvStore {
+        KvStore {
+            data: Arc::new(RwLock::new(HashMap::new())),
+        }
+    }
+
+    pub fn raw_get(&self, key: &[u8]) -> Vec<u8> {
+        let data = self.data.read().unwrap();
+        data.get(key).unwrap_or(&vec![]).to_vec()
+    }
+
+    pub fn raw_batch_get(&self, keys: &[Vec<u8>]) -> Vec<KvPair> {
```

Comment:
I think you can take `keys: Vec<Vec<u8>>` then use `into_iter` and `key` instead of `key.to_vec()`.